provider/aws/lambda/cfevents/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+)
+
+var (
+	cf  = cloudformation.New(session.New(), nil)
+	cwl = cloudwatchlogs.New(session.New(), nil)
+)
+
+// Event represents a lambda event message
+type Event struct {
+	Records []Record
+}
+
+// Record is an entry in a lambda event
+type Record struct {
+	Sns struct {
+		Message   string
+		MessageId string
+	}
+}
+
+// Message is a map of data sent in a Cloudformation event
+type Message map[string]string
+
+func main() {
+	if len(os.Args) < 2 {
+		die(fmt.Errorf("must specify event as argument"))
+	}
+
+	data := []byte(os.Args[1])
+
+	var e Event
+
+	if err := json.Unmarshal(data, &e); err != nil {
+		die(err)
+	}
+
+	for _, r := range e.Records {
+		if err := handle(r); err != nil {
+			die(err)
+		}
+	}
+}
+
+func handle(r Record) error {
+	m, err := parseMessage(r.Sns.Message)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("m = %+v\n", m)
+
+	resp, err := cf.DescribeStacks(&cloudformation.DescribeStacksInput{
```

Comment:
we've had some problems in the past with these types of calls triggering rate limiting.